File: Singer.API/DTOs/EventRegistrationLogDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Singer.DTOs
{
   public class EventRegistrationLogDTO
   {
      public Guid Id { get; set; }
      public Guid EventRegistrationId { get; set; }
      public string EventTitle { get; set; }
      public string CareUser { get; set; }
      public List<string> LegalGuardians { get; set; }
      public DateTime CreationDateTimeUTC { get; set; }
      public DateTime EventSlotStartDateTime { get; set; }
      public DateTime EventSlotEndDateTime { get; set; }
   }
}

File: Singer.API/ClientApp/src/app/modules/core/DTOs/event-registration-log.dto.ts
import { RegistrationStatus } from '../models/enum';

export interface RegistrationLogCareUserDTO {
   id: string;
   careUser: string;
   legalGuardians: LegalGuardianDTO[];
   creationDateTimeUTC: Date;
   registrationStateChanges: CareUserRegistrationStateChangedDTO[];
   registrationLocationChanges: CareUserRegistrationLocationChangedDTO[];
}

export interface RegistrationLogCareUserRegistrationDTO {
   eventRegistrationId: string;
   eventTitle: string;
   eventSlotStartDateTime: Date;
   eventSlotEndDateTime: Date;
}
export interface LegalGuardianDTO {
   name: string;
   email: string;
}
export interface CareUserRegistrationStateChangedDTO {
   eventRegistrationId: string;
   eventTitle: string;
   eventSlotStartDateTime: Date;
   eventSlotEndDateTime: Date;
   newStatus: RegistrationStatus;
}
export interface CareUserRegistrationLocationChangedDTO {
   eventRegistrationId: string;
   eventTitle: string;
   eventSlotStartDateTime: Date;
   eventSlotEndDateTime: Date;
   newLocation: string;
}
